### PR TITLE
ERR: Raise ValueError when non-default index is given for orc format

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

orc can't handle indexes right now (they are dropped), so we should raise like we did in feather

### CODE REVIEW

File: pandas/io/orc.py [hunk 1]
```diff
@@ -209,6 +210,21 @@ def to_orc(
     if engine_kwargs is None:
         engine_kwargs = {}
 
+    # validate index
+    # --------------
+
+    # validate that we have only a default index
+    # raise on anything else as we don't serialize the index
+
+    if not df.index.equals(RangeIndex.from_range(range(len(df)))):
```
Comment (mroeschke) [inline pandas/io/orc.py:219]: Might be clearer to use `pandas.core.indexes.api.default_index`

**Comment by mroeschke** [comment]
> orc can't handle indexes right now (they are dropped)

Just confirming, they are dropped without a warning/exception from pyarrow?

File: pandas/io/orc.py (hunk 1 above)
Comment (phofl) [reply pandas/io/orc.py]: Updated

**Comment by phofl** [comment]
Yes they are just gone when roundtripping

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -174,7 +174,7 @@ MultiIndex
 
 I/O
 ^^^
--
+- :meth:`DataFrame.to_orc` now raising ``ValueError`` when non-default :class:`Index` is given (:issue:`0`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:177]:
```suggestion
- :meth:`DataFrame.to_orc` now raising ``ValueError`` when non-default :class:`Index` is given (:issue:`51828`)
```

**Review by mroeschke — COMMENTED** [review]
Merge conflict too otherwise looks good

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**